src/amr/tagging/default_hybrid_tagger_strategy.h

```diff
@@ -0,0 +1,79 @@
+#ifndef DEFAULT_HYBRID_TAGGER_STRATEGY_H
+#define DEFAULT_HYBRID_TAGGER_STRATEGY_H
+
+#include "hybrid_tagger_strategy.h"
+#include "core/data/grid/gridlayoutdefs.h"
+#include "core/data/vecfield/vecfield_component.h"
+
+
+namespace PHARE::amr
+{
+template<typename HybridModel>
+class DefaultHybridTaggerStrategy : public HybridTaggerStrategy<HybridModel>
+{
+    using gridlayout_type           = typename HybridModel::gridlayout_type;
+    static auto constexpr dimension = HybridModel::dimension;
+
+public:
+    void tag(HybridModel& model, gridlayout_type const& layout, int* tags) const override;
+};
+
+template<typename HybridModel>
+void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
+                                                   gridlayout_type const& layout, int* tags) const
+{
+    auto& Bx = model.state.electromag.B.getComponent(PHARE::core::Component::X);
+    auto& By = model.state.electromag.B.getComponent(PHARE::core::Component::Y);
+    auto& Bz = model.state.electromag.B.getComponent(PHARE::core::Component::Z);
+
+    auto& N = model.state.ions.density();
+
+    // we loop on cell indexes for all qties regardless of their centering
+    auto start
+        = layout.physicalStartIndex(PHARE::core::QtyCentering::dual, PHARE::core::Direction::X);
+    auto end = layout.physicalEndIndex(PHARE::core::QtyCentering::dual, PHARE::core::Direction::X);
+
+    auto endCell     = layout.nbrCells()[0] - 1;
+    double threshold = 0.05;
+    int idx          = 4;
+    double epsilon   = 0.1;
+
+
+    if constexpr (dimension == 1)
+    {
+        for (auto iCell = 0u, ix = start; iCell <= endCell; ++ix, ++iCell)
+        {
+            auto Byavgm2 = 0.2 * (By(ix - 4) + By(ix - 3) + By(ix - 2) + By(ix - 1) + By(ix));

```

Comment:
is there some documentation to go along with this as a reference?

it's pretty complicated